src/parser/core/modules/CastTime.ts

```diff
@@ -164,18 +211,20 @@ export default class CastTime extends Analyser {
 
 		// Mimicking old logic w/ the undefined. Don't ask.
 		const action = this.data.getAction(actionId)
-		const defaultCastTime = action?.castTime
+		const defaultTime = forRecast ? (action?.gcdRecast != null
+			? action?.gcdRecast
+			: action?.cooldown) : action?.castTime
 
 		// If there were no modifiers, just use the default (or if the default comes back undefined or already instant, shouldn't happen but eh)
-		if (!matchingTimes.length || defaultCastTime == null || defaultCastTime === 0) {
-			return defaultCastTime
+		if (!matchingTimes.length || defaultTime == null || defaultTime === 0) {
+			return defaultTime
 		}
-
 		let flatReduction=0
 		let flatIncrease=0
 		let percentageAdjustment=1
 
 		matchingTimes.forEach(ct => {
+			if (!ct.affectsRecast && forRecast) { return }
```

Comment:
can probably be moved into the original filter so that the short circuit can pick it up

Reply:
Good idea. Hopefully I didn't totally brainfuzz the boolean logic there...

Reply:
Definitely no brainfuzz now that it's not a boolean parameter anymore :)